### PR TITLE
MM New derived table Holdings tags

### PR DESCRIPTION
**Pull request by natalyapik**
create new metadb derived table holdings_tags

### CODE REVIEW

**Comment by nassibnassar** [comment]
- [ ] Pull request has a complete description
- [ ] Query runs without errors
- [ ] Query output is correct
- [ ] Query logic is clear and well documented
- [ ] Query is readable and properly indented
- [ ] Table and column names are in all-lowercase
- [ ] Quotation marks are used only where necessary
- [ ] Query begins with user documentation in comment lines
- [ ] File name is listed in `runlist.txt` after dependencies
- [ ] All columns have indexes
- [ ] Table is vacuumed and analyzed

File: sql_metadb/derived_tables/holdings_tags [hunk 1]
```diff
@@ -0,0 +1,27 @@
+-- This derived table extracts tags from the holdings record.
+DROP TABLE IF EXISTS holdings_tags;
+
+CREATE TABLE holdings_tags AS 
+SELECT 
+    hd.instance_id,
+    hd.id AS holdings_id,
+    hd.hrid AS holdings_hrid,
+    hldtags.jsonb #>> '{}' AS holdings_tags,
```
Comment (jenmawe) [inline sql_metadb/derived_tables/holdings_tags:9]: This column name should probably be holdings_tags. I changed it for the instance tag column name to be singular. This is just to follow that all column names be singular.

File: sql_metadb/derived_tables/holdings_tags [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    hldtags.ordinality AS holdings_tags_ordinality
```
Comment (jenmawe) [inline sql_metadb/derived_tables/holdings_tags:10]: To follow the comment above, change tags to tag here

**Review by jenmawe — CHANGES_REQUESTED** [review]

**Review by adoerrer — CHANGES_REQUESTED** [review]
* [ x]  Pull request has a complete description
     * [x]  Query runs without errors
     * [x]  Query output is correct
     * [x ]  Query logic is clear and well documented
     * [x]  Query is readable and properly indented
     * [x]  Table and column names are in all-lowercase 
     * [x]  Quotation marks are used only where necessary 
     * [x]  Query begins with user documentation in comment lines 
     * [x]  File name is listed in `runlist.txt` after dependencies 
     * [x]  All columns have indexes
     * [x]  Table is vacuumed and analyzed

Thank you, Natalya! Although the query runs great and checklist is all checked there are some minor change requests.

As Jennifer already mentioned please change the column name regarding tags to singular. And the query file is missing the `.sql ` suffix.

**Review by adoerrer — APPROVED** [review]
Perfect, columnnames look good now. Thank you, Natalya!

File: sql_metadb/derived_tables/holdings_tags.sql [hunk 3]
```diff
@@ -0,0 +1,27 @@
+-- This derived table extracts tags from the holdings record.
+DROP TABLE IF EXISTS holdings_tags;
+
+CREATE TABLE holdings_tags AS 
+SELECT 
+    hd.instance_id,
+    hd.id AS holding_id,
```
Comment (jenmawe) [inline sql_metadb/derived_tables/holdings_tags.sql:7]: For MM, we still want to use holdings. This is how we use it in our other queries.

File: sql_metadb/derived_tables/holdings_tags.sql [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    hd.hrid AS holding_hrid,
```
Comment (jenmawe) [inline sql_metadb/derived_tables/holdings_tags.sql:8]: holdings_hrid

File: sql_metadb/derived_tables/holdings_tags.sql [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    hldtags.jsonb #>> '{}' AS holding_tag,
```
Comment (jenmawe) [inline sql_metadb/derived_tables/holdings_tags.sql:9]: holdings_tag

File: sql_metadb/derived_tables/holdings_tags.sql [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    hldtags.ordinality AS holding_tag_ordinality
```
Comment (jenmawe) [inline sql_metadb/derived_tables/holdings_tags.sql:10]: holdings_tag_ordinality

File: sql_metadb/derived_tables/holdings_tags.sql [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+FROM 
+    folio_inventory.holdings_record__t AS hd
+    LEFT JOIN folio_inventory.holdings_record AS hdr ON hdr.id = hd.id
+    CROSS JOIN LATERAL jsonb_array_elements(jsonb_extract_path(hdr.jsonb, 'tags', 'tagList'))
+    WITH ORDINALITY AS hldtags (jsonb);
+    
+CREATE INDEX ON holdings_tags (instance_id);
+
+CREATE INDEX ON holdings_tags (holdins_id);
```
Comment (jenmawe) [inline sql_metadb/derived_tables/holdings_tags.sql:19]: missing the g for holdings_id

File: sql_metadb/derived_tables/holdings_tags.sql [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+CREATE INDEX ON holdings_tags (holding_hrid);
```
Comment (jenmawe) [inline sql_metadb/derived_tables/holdings_tags.sql:21]: holdings_hrid and the same for those below

**Review by jenmawe — CHANGES_REQUESTED** [review]
Hi @natalyapik ,

We did decide to use holdings rather than holding in MM as we refer to a holdings record. Anyway, this one still uses the plural but the term that follows won't like holdings_tag or holdings_note. Sorry for the confusion.

**Review by adoerrer — APPROVED** [review]
Thanks @jenmawe  for catching the spelling issues. I think this is now ready to go.

@jenmawe Do you want to re-review the changes as well?  Thanks!

**Review by jenmawe — APPROVED** [review]

**Pull request merged**